src/highdicom/content.py

```diff
@@ -936,3 +1067,536 @@ def __init__(
+                    )
+                self.WindowWidth = [
+                    format_number_as_ds(x) for x in window_width
+                ]
+            else:
+                if window_is_sequence:
+                    raise TypeError(
+                        'Length of "window_width" must match length of '
+                        '"window_center".'
+                    )
+                self.WindowWidth = format_number_as_ds(window_width)
+        if window_explanation is not None:
+            if window_center is None:
+                raise TypeError(
+                    'Providing "window_explanation" is invalid if '
+                    '"window_center" is not provided.'
+                )
+            if isinstance(window_explanation, str):
+                if window_is_sequence:
+                    raise TypeError(
+                        'Length of "window_explanation" must match length of '
+                        '"window_center".'
+                    )
+                _check_long_string(window_explanation)
+            elif isinstance(window_explanation, Sequence):
+                if (
+                    not window_is_sequence or
+                    (len(window_explanation) != len(window_center))
+                ):
+                    raise ValueError(
+                        'Length of "window_explanation" must match length of '
+                        '"window_center".'
+                    )
+                if len(window_explanation) == 0:
+                    raise TypeError(
+                        'Argument "window_explanation" must not be an empty '
+                        'sequence.'
+                    )
+                for exp in window_explanation:
+                    _check_long_string(exp)
+            self.WindowCenterWidthExplanation = window_explanation
+        if voi_lut_function is not None:
+            if window_center is None:
+                raise TypeError(
+                    'Providing "voi_lut_function" is invalid if '
+                    '"window_center" is not provided.'
+                )
+            self.VOILUTFunction = VOILUTFunctionValues(voi_lut_function).value
+
+        if voi_luts is not None:
+            if len(voi_luts) == 0:
+                raise ValueError('"voi_luts" should not be empty.')
+            for lut in voi_luts:
+                if not isinstance(lut, LUT):
+                    raise TypeError(
+                        'Items of "voi_luts" should be of type highdicom.LUT.'
+                    )
+            self.VOILUTSequence = list(voi_luts)
+        else:
+            if window_center is None:
+                raise TypeError(
+                    'At least one of "window_center" or "voi_luts" should be '
+                    'provided.'
+                )
+
+
+class PaletteColorLookupTable(Dataset):
+
+    """Dataset describing a palette color lookup table."""
+
+    def __init__(
```

Comment:
I wonder whether we should pass `red_lut: LUT`, `green_lut: LUT`, `blue_lut: LUT` and reuse the `LUT` abstraction?

Reply:
I was initially opposed to this because the names of the attributes change when they are copied from each `LUT` to the destination attribute. However I think this is more convenient for both users and for us to maintain so I've switched to this as suggested.

Reply:
Great minds think alike: https://github.com/herrmannlab/highdicom/pull/139#discussion_r839580863

Reply:
Yeah, just saw your comment 👍

Reply:
I am afraid the envisioned `SegmentedLUT` approach may not work, because there is no such thing in DICOM. Segmented Lookup Table Data appears to be only used in Palette Color Lookup Table. A `SegmentedLUTData` attribute does not exist.

Reply:
I would suggest implementing two classes:
- `PaletteColorLUT`
- `SegmentedPaletteColorLUT`

They can both have the same interface as `LUT`.

Reply:
See bb6129d110db6a9c3a1d20f2414a0f72fb4bdac6

Reply:
Addressed via https://github.com/herrmannlab/highdicom/pull/166